File: src/main/java/github/jk1/smtpidea/server/smtp/SmtpsMailServer.java
package github.jk1.smtpidea.server.smtp;

import javax.net.ssl.SSLServerSocketFactory;
import java.io.IOException;
import java.net.InetSocketAddress;
import java.net.ServerSocket;

/**
 * TLS-secured SMTP server implementation.
 * It requires client connections to start with a handshake.
 *
 * @author Evgeniy Naumenko
 */
class SmtpsMailServer extends SmtpMailServer {

    /**
     * @param configuration
     */
    public SmtpsMailServer(ServerConfiguration configuration) {
        super(configuration);
        this.setEnableTLS(false); // disable STARTTLS, it makes no sense here
    }

    /**
     * {@inheritDoc}
     */
    @Override
    protected ServerSocket createServerSocket() throws IOException {
        InetSocketAddress isa = new InetSocketAddress(this.getBindAddress(), this.getPort());
        SSLServerSocketFactory factory = (SSLServerSocketFactory) SSLServerSocketFactory.getDefault();
        ServerSocket serverSocket = factory.createServerSocket(this.getPort(), this.getBacklog(), isa.getAddress());
        if (this.getPort() == 0) {
            this.setPort(serverSocket.getLocalPort());
        }
        return serverSocket;
    }
}

File: src/main/kotlin/github/jk1/smtpidea/server/smtp/SmtpsMailServer.kt
package github.jk1.smtpidea.server.smtp

import github.jk1.smtpidea.config.SmtpConfig
import java.net.ServerSocket
import java.net.InetSocketAddress
import javax.net.ssl.SSLServerSocketFactory

/**
 * TLS-secured SMTP server implementation.
 * It requires client connections to start with a handshake.
 *
 * @author Evgeniy Naumenko
 */
public class SmtpsMailServer(configuration: SmtpConfig) : SmtpMailServer(configuration){

    {
        setEnableTLS(false); // disable STARTTLS, it makes no sense here
    }

    protected override fun createServerSocket(): ServerSocket {
        val isa = InetSocketAddress(getBindAddress(), getPort())
        val factory = SSLServerSocketFactory.getDefault() as SSLServerSocketFactory
        val serverSocket = factory.createServerSocket(getPort(), getBacklog(), isa.getAddress())
        if (getPort() == 0) setPort(serverSocket.getLocalPort())
        return serverSocket
    }
}
